File: FamilyTools.EasyCompta/Model/AccountPage.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace FamilyTools.EasyCompta.Model
{
    [Table("AccountPages")]
    public class AccountPage : BaseModel
    {
        public string Name { get; set; }

        public List<AccountEnter> Enters { get; set; }

        [NotMapped]
        public Dictionary<User, bool> PaymentDone { get; set; }

        public bool IsClosing { get; set; }

        public DateTime Date { get; set; }

        public float Total { get; set; }

        public AccountPage()
        {
            if (Enters?.Count > 0)
            {
                Total = Enters.Sum(enter => enter.TotalValue);

                PaymentDone ??= [];

                PaymentDone = Enters.SelectMany(enter => enter.Lines)
                        .Select(line => line.UserLink)
                        .Distinct()
                        .ToDictionary(
                        user => user,
                        user => PaymentDone != null && PaymentDone.ContainsKey(user) ? PaymentDone[user] : false
                    );
            }
        }
    }
}

File: FamilyTools.Web/src/app/models/account-page.ts
import { BaseModel } from './base-model';
import { User } from './user';
import { AccountEnter } from './account-enter';

export class AccountPage extends BaseModel {
  name: string;
  enters: AccountEnter[];
  paymentDones: Map<User, boolean>;
  isClosing: boolean;
  date: Date;

  constructor(  name: string, enters: AccountEnter[], paymentDones: Map<User, boolean>, isClosing: boolean, date: Date, id?: number, creationDate?: Date, updateDate?: Date){
    super(id, creationDate, updateDate);
    this.name = name;
    this.enters = enters;
    this.paymentDones = paymentDones;
    this.isClosing = isClosing;
    this.date = date;
  }

    getUsers(): User[] {
      return Array.from(this.paymentDones.keys());
    }

    getTotalForUser(id: number){
      this.enters.map(enters => enters.lines.filter(lines => lines.userLink.id == id).reduce((acc, lines) => acc + lines.value, 0))
    }

    getTotal(){
      this.enters.reduce((acc, enter) => acc + enter.totalValue, 0)
    }
}
